src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,383 @@
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'Number of users to process at once',
+                self::DEFAULT_BATCH_SIZE
+            )
+            ->addOption(
+                'password-ttl',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'After how many days passwords expire. Set when Content Type needs to be updated.',
+                self::DEFAULT_PASSWORD_TTL
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> expires passwords of specific users. 
+Use this tool wisely as next time affect users tries to log in, they will be forced to set a new password.
+Note: This script can potentially run for a very long time, and in Symfony dev environment it will consume memory exponentially with size of dataset.
+
+{$beforeRunningHints}
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $batchSize = $input->getOption('batch-size');
+        $userIds = $input->getOption('user-id');
+        $userGroupIds = $input->getOption('user-group-id');
+        $force = $input->getOption('force');
+        $passwordTTL = $input->getOption('password-ttl');
+
+        if (!empty($userIds) && !empty($userGroupIds)) {
+            throw new InvalidArgumentException('You cannot use --user-id and --user-group-id options at once.');
+        }
+
+        $userContentTypeIdentifiers = $input->getOption('user-content-type-identifier');
+
+        $this->summarizeSearchCriteria($output, $userIds, $userGroupIds, $userContentTypeIdentifiers);
+
+        $criteria = $this->getCriteria($userIds, $userGroupIds, $userContentTypeIdentifiers);
+
+        $filter = new Filter(new LogicalAnd($criteria));
+        $filter->withLimit(0);
+
+        $contentList = $this->repository->sudo(static function (Repository $repository) use ($filter): ContentList {
+            return $repository->getContentService()->find($filter);
+        });
+        $count = $contentList->getTotalCount();
+
+        if ($count === 0) {
+            $output->writeln('<info>There are no users matching given criteria</info>');
+
+            return Command::SUCCESS;
+        }
+
+        $output->writeln(sprintf(
+            '<info>Found %d User(s) matching given criteria</info>',
+            $count
+        ));
+
+        $displayProgressBar = !($output->isVerbose() || $output->isVeryVerbose() || $output->isDebug());
+
+        if ($displayProgressBar) {
+            $progressBar = new ProgressBar($output, $count);
+            $progressBar->setFormat(
+                '%current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%' . PHP_EOL
+            );
+            $progressBar->start();
+        }
+
+        $processedUsersCount = 0;
+        $processedContentTypes = [];
+
+        $this->repository->beginTransaction();
+        try {
+            do {
```

Comment:
It feels like the content of `do ... while` loop can be extracted to at least one method, it's hardly readable to me in the current form tbh.